client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/evaluation/softclassification/BinarySoftClassification.java

```diff
@@ -88,7 +94,10 @@ public BinarySoftClassification(String actualField, String predictedProbabilityF
                                     @Nullable List<EvaluationMetric> metrics) {
         this.actualField = Objects.requireNonNull(actualField);
         this.predictedProbabilityField = Objects.requireNonNull(predictedProbabilityField);
-        this.metrics = Objects.requireNonNull(metrics);
+        if (metrics != null) {
```

Comment:
What does it mean if there are no Evaluation Metrics, does that make any sense to allow this?

Reply:
Either way this is an improvement on the code that tagged `metrics` as nullable then has `requireNonNull(metrics)` 3 lines later

Reply:
Oh I see now. The server side has a set of default metrics that are used if this parameter is null

Reply:
yeah, `null` => `use default value`. This follows the implementation pattern we have elsewhere in the HLRC.